src/components/ScrollLock/ScrollLock.tsx

```diff
@@ -1,21 +1,24 @@
 import * as React from 'react';
 import './ScrollLock.scss';
-import {polarisAppProviderContextTypes} from '../AppProvider';
+import {WithAppProviderProps, withAppProvider} from '../AppProvider';
 
-export default class ScrollLock extends React.Component<{}, never> {
-  static contextTypes = polarisAppProviderContextTypes;
+export interface Props {}
 
+type CombinedProps = Props & WithAppProviderProps;
+class ScrollLock extends React.Component<CombinedProps, never> {
   componentDidMount() {
-    const {scrollLockManager} = this.context.polaris;
+    const {scrollLockManager} = this.props.polaris;
```

Comment:
I thought the `polaris` namespace was removed...

Reply:
Partially, our context was namespaced. So whenever we consumed context it was `this.context.polaris[X]`. If we migrated to hooks it would have been
```tsx
const polaris = React.useContext(AppProviderContext);
const {translate} = polaris.polaris.intl;
```

We're still using `polaris` when we pass down our app provider props, however now we don't have a double polaris to account for. And when we use hooks moving forward we can do

```tsx
const {intl, X} = React.useContext(AppProviderContext);
```